src/validation_workflow/download_utils.py

```diff
@@ -0,0 +1,28 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+
+import requests
+
+from system.temporary_directory import TemporaryDirectory
+
+
+class DownloadUtils:
+    @staticmethod
+    def is_url_valid(url: str) -> int:
+        response = requests.head(url)
+        if response.status_code == 200:
+            return 1
+        else:
+            return 0
+
+    @staticmethod
+    def download(url: str, tmp_dir: TemporaryDirectory) -> bool:
```

Comment:
nit: rename to download_artifacts
Just download is bit confusing

Reply:
This is a different function in downloadutils.py which is called by the download_artifacts method in tar, rpm and yum.